sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -210,6 +220,12 @@ class DataFrame private[sql](
     }
 
     sb.append(sep)
+
+    // For Data that has more than N(numRows) records
+    if (take(numRows + 1).length > numRows) {
```

Comment:
don't call take again. We've already called take once in the beginning of the function.